src/signal/cr_signal.cc

```diff
@@ -6,6 +6,7 @@
 #if defined(__clang__) || defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wconversion"
+#pragma GCC diagnostic ignored "-Wshadow"
```

Comment:
Why?

Reply:
Because the boost headers trigger the warning.

Reply:
FYI I often describe this in comment on top of these lines, probably also with the version of boost with this issue.
When header comes and goes, people don't always remember to verify/remove this, so legacy will be left behind, till the point of "who did this? it doesn't make sense, I cannot repro it."